framework/db/ActiveRecord.php

```diff
@@ -455,18 +455,28 @@ protected function insertInternal($attributes = null)
             }
         }
         $db = static::getDb();
-        $command = $db->createCommand()->insert($this->tableName(), $values);
-        if (!$command->execute()) {
+        // depending on the underlying DBMS, the command may return results
+        // or lastInsertId has to be used to fetch primary key values
+        $command = $db->createCommand()->insertReturning($this->tableName(), $values, $this->primaryKey());
+        // force preparing as a command for writing, not reading
+        $command->prepare(false);
+        $primaryKeys = $command->queryOne();
```

Comment:
https://github.com/nineinchnick/yii2/blob/5042-insert-returning/framework/db/Command.php#L880

`Command::_returningParams` defaults to null, so `queryInternal` behavior doesn't change and `pgsql\QueryBuilder` sets it to `true` which also restores default behavior of `queryInternal`.

Default `insertReturning` sets it to an empty array, which makes `queryInternal` iterate over it, which does nothing (no fetching).